### PR TITLE
feat (operator): get verification data batch from cloudfront

### LINKED ISSUE #154: feat (operator): get batch from Cloudflare
# Description
- Get verification data batch from cloudflare

### PR DESCRIPTION
**Pull request by taturosati**
# Description

- Operator gets verification data from Cloudflare instead of from fs

# To test

1. Have anvil, aggregator & operator running
2. Run `make send_plonk_bls12_381_proof`
3. Operator should get batch from s3

You can see file content yourself by downloading batch file from [here](https://storage.alignedlayer.com/b4b654a31b43c7b5711206eea7d44f884ece1fe7164b478fa16215be77dc84cb.json)

### CODE REVIEW

File: batcher/test-client/src/main.rs [hunk 1]
```diff
@@ -57,15 +57,15 @@ async fn upload_proof_to_s3() -> Option<Duration> {
             .expect("Failed to read proof file");
 
     let client = batcher::s3::create_client().await;
-    let bytes = bytes::Bytes::from(proof);
+    // let bytes = bytes::Bytes::from(proof);
```
Comment (entropidelic) [inline batcher/test-client/src/main.rs]:
```suggestion
```

File: common/proving_systems.go [hunk 2]
```diff
@@ -8,3 +10,26 @@ const (
...
+	return [...]string{"GnarkPlonkBls12_381", "GnarkPlonkBn254", "Groth16Bn254", "SP1"}[*t]
+}
+
+func (t *ProvingSystemId) FromString(provingSystem string) ProvingSystemId {
+	return map[string]ProvingSystemId{
+		"GnarkPlonkBls12_381": GnarkPlonkBls12_381,
+		"GnarkPlonkBn254":     GnarkPlonkBn254,
+		"Groth16Bn254":        Groth16Bn254,
+		"SP1":                 SP1,
+	}[provingSystem]
```
Comment (entropidelic) [inline common/proving_systems.go]: I'm wondering if creating the map every time the function is called could be a problem later on. Maybe we can define the map as a global variable with `var` or use `switch`?

Comment (entropidelic) [reply common/proving_systems.go]: also, it seems like the reference `t` is not being used, could it be removed?

File: operator/pkg/s3.go [hunk 3]
```diff
@@ -1,43 +1,37 @@
...
-
-	// proof, err := io.ReadAll(resp.Body)
-	// if err != nil {
-	// 	return nil, err
-	// }
-
-	// FIXME(marian): We are simulating getting the batch from s3, but this is hardcoded right now
-	proofBytes, err := os.ReadFile("./task_sender/test_examples/gnark_plonk_bn254_script/plonk.proof")
+func (o *Operator) getBatchFromS3(proofUrl string) ([]VerificationData, error) {
+	log.Println("Getting batch from S3..., proofUrl:", proofUrl)
```
Comment (entropidelic) [inline operator/pkg/s3.go]: small nit, but let's use the operator logger here
```suggestion
	o.Logger.Infof(...)
```

Comment (taturosati) [reply operator/pkg/s3.go]: Changed on 9ca7d24706c933d6234160a67d872e6842f40443

File: common/proving_systems.go (hunk 2 above)
Comment (taturosati) [reply common/proving_systems.go]: Changed on 9ca7d24706c933d6234160a67d872e6842f40443

File: task_sender/cmd/main.go [hunk 4]
```diff
@@ -128,11 +128,11 @@ func taskSenderMain(c *cli.Context) error {
...
 	// TODO(marian): Remove this hardcoded merkle root
 	var batchMerkleRoot [32]byte
-	batchMerkleRoot[0] = byte(2)
-	batchMerkleRoot[1] = byte(3)
+	batchMerkleRoot[0] = byte(123)
+	batchMerkleRoot[1] = byte(123)
 
 	// TODO(marian): Remove this dummy S3 url
-	batchDataPointer := "aligned.awesome.batch.s3.com"
+	batchDataPointer := "https://storage.alignedlayer.com/b4b654a31b43c7b5711206eea7d44f884ece1fe7164b478fa16215be77dc84cb.json"
```
Comment (uri-99) [inline task_sender/cmd/main.go:135]: Is this the hash of the proof bytes?
I'm thinking i can add this dinamically in [this pr line](https://github.com/yetanotherco/aligned_layer/pull/177/files#diff-d588c90fe84227a6aac51d03e028b326a66663f0849302f69d7f1fd9c553dbe0R201)

**Review by entropidelic — APPROVED** [review]

**Review by NicolasRampoldi — APPROVED** [review]
lgtm. Tested it and it works.

**Pull request merged**